velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -1383,6 +1385,85 @@ TpchPlan TpchQueryBuilder::getQ16Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ17Plan() const {
+  std::vector<std::string> lineitemColumns = {
+      "l_partkey", "l_extendedprice", "l_quantity"};
+  std::vector<std::string> partColumns = {
+      "p_partkey", "p_brand", "p_container"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto partSelectedRowType = getRowType(kPart, partColumns);
+  const auto& partFileColumns = getFileColumnNames(kPart);
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemScanNodeId;
```

Comment:
nit: drop 'Node' for brevity

nit: The code is a bit difficult to read. I understand there are 2 identical joins between lineitem and filtered part. Consider adding a lambda to produce that plan and fill in pair of scan node IDs. Also, make sure that lineitemScanId and partScanId point to nodes in one subquery and lineitemAggScanId and partAggScanId point to nodes in another subquery. As written, it seems that lineitemScanNodeId and partScanNodeIdSubQuery go together, which is confusing.

lineitemScanId, partScanId, lineitemAggScanId, partAggScanId

Reply:
Thanks for the suggestion, updated scanIds accordingly. Writing a lambda would require the optimal join ordering to be changed and the average aggregation would need to be calculated after the join for the subquery as well, which seems cumbersome.